indra/databases/identifiers.py

```diff
@@ -278,11 +277,59 @@ def parse_identifiers_url(url):
     return None, None
 
 
+def namespace_embedded(db_ns: str) -> bool:
+    """Return true if this namespace requires IDs to have namespace embedded.
+
+    This function first maps the given namespace to an identifiers.org
+    namespace and then checks the registry to see if namespaces need
+    to be embedded in IDs. If yes, it returns True. If not, or the ID can't
+    be mapped to identifiers.org, it returns False
+
+    Parameters
+    ----------
+    db_ns :
+        The namespace to check.
+
+    Returns
+    -------
+    :
+        True if the namespace is known to be embedded in IDs of this
+        namespace. False if unknown or known not to be embedded.
+    """
+    identifiers_ns = get_identifiers_ns(db_ns)
+    if identifiers_ns:
+        identifiers_entry = identifiers_registry.get(identifiers_ns)
+        if identifiers_entry['namespace_embedded']:
+            return True
+    return False
+
+
+def ensure_prefix_if_needed(db_ns: str, db_id: str) -> str:
+    """Return an ID ensuring a namespace prefix if known to be needed.
+
+    Parameters
+    ----------
+    db_ns :
+        The namespace associated with the identifier.
+    db_id :
+        The original identifier.
+
+    Returns
+    -------
+    :
+        The identifier with namespace embedded if needed.
```

Comment:
@cthoyt I found that this way of annotating the return type renders correctly and doesn't require naming a return variable or duplicating the already specified return type.